ls1intum/Artemis
# `General`: Show invalid input in forms related to user management

### Checklist
#### General


#### Client

### Motivation and Context



Closes #4916 (in addition, i also checked if the server validates the length -> looks good)
And as a user/admin, it is sometimes difficult to know the limitations of some inputs. Therefore, this PR shows the requirements of some input fields when adding/creating new users.

### Description

In order to validate the inputs and show the results to the user, we need to use a `formGroup` which contains all inputs (`editForm` in [user-management-update.component.ts](https://github.com/ls1intum/Artemis/pull/5018/files#diff-df49c8bbcd95ce6b0695f83932ab705c08f4adda416eac41fc1868f2c9257366)). In addition, I also extracted some hard coded values in variables (e.g. `EMAIL_MIN_LENGTH`, ...). In order to highlight the required fields, we needed to add `required` to the `<inputs/>`s and use the validators.

### Steps for Testing

Prerequisites:
- 1 Admin

1. Log in to Artemis
2. Navigate to User Management (`admin/user-management`)
3. Create a new User and verify the required fields (e.g. firstName/lastName is required by Bitbucket)
4. Navigate to Password Settings (`/account/password`)
5. Check if the maximum length works as expected 

### Review Progress





#### Code Review
#### Manual Tests

### Test coverage
_not changed_
| Class/File | Branch | Line |
|------------|-------:|-----:|
| user-management-update.component.ts | 68.33% |  91.95% |

### Screenshots



<details>
<summary>Videos</summary>

https://user-images.githubusercontent.com/60005702/166658221-c37953f5-5636-4567-9d61-f7699c357042.mp4

https://user-images.githubusercontent.com/60005702/166658504-75fc370f-c915-4c67-94e5-38a7c8859efb.mp4
</details>
